File: core/documents/document_dependency_graph_engine.py
from __future__ import annotations

from typing import Any, Dict

from core.documents.instructional_flow_engine import extract_instructional_flow
from core.documents.concept_transition_engine import model_concept_transitions


def build_document_dependency_graph(text: str) -> Dict[str, Any]:
    flow = extract_instructional_flow(text)
    trans = model_concept_transitions(text)
    nodes = [{"id": s.get("title", f"step{i}"), "kind": "step"} for i, s in enumerate(flow["steps"])]
    edges = trans.get("transitions", [])
    return {"nodes": nodes, "edges": edges, "prerequisites": flow.get("prerequisites", [])}

File: src/documents/documentDependencyGraphEngine.ts
/**
 * Converted from Python: core/documents/document_dependency_graph_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { extractInstructionalFlow } from "./instructionalFlowEngine.js";
import { modelConceptTransitions } from "./conceptTransitionEngine.js";

export function buildDocumentDependencyGraph(text: any): any {
  var flow: any = extractInstructionalFlow(text);
  var trans: any = modelConceptTransitions(text);
  var nodes: any = py.enumerate(py.at(flow, "steps")).map(([i, s]: any) => ({"id": py.get(s, "title", `step${py.toStr(i)}`), "kind": "step"}));
  var edges: any = py.get(trans, "transitions", []);
  return {"nodes": nodes, "edges": edges, "prerequisites": py.get(flow, "prerequisites", [])};
}
export { extractInstructionalFlow, modelConceptTransitions };
